packages/x-charts/src/hooks/useSeries.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { SeriesContext } from '../context/SeriesContextProvider';
+
+/**
+ * Get access to the internal state of series.
+ * Structured by type of series:
+ * { seriesType?: { series: { id1: precessedValue, ... }, seriesOrder: [id1, ...] } }
+ * @returns FormattedSeries series
+ */
```

Comment:
> I see this is unstable though, we can probably figure out a better type/output for this eventually 😅

Yes, unstable does not mean we do crap, it only means we are unsure it will stay like that for a year
When you say about type/output do you have some specific ideas?

Reply:
I'm not sure if it is useful to return an object with `{bar?: {}, pie?: {}, ...etc}`. Maybe I'm missing the use-case, but wouldn't it be clearer for the user if we divided them up instead, like `usePieSeries()`?

Reply:
I considered the `usePieSeries`, `useBarSeries`, `useLineSeries`, ...
So it might be a good idea if you think about it too 😁

For the typing, it's because we have a stack of type inheritance with the series type as generic. So the `FormattedSeries` is in fact something like `{[type in seriesType]?: SeriesFormatterResult<type> }`

I'm curious about how it will scale when adding heatmap, radars, and other charts that are more independent from each other.

Maybe we will realise that the tradeof between leveraging TS generics vs copy pasting for type simplicity is not in favor of the current config solution

Reply:
Also on another note, is it currently possible to have two charts of the same type in the graph?

Reply:
> Also on another note, is it currently possible to have two charts of the same type in the graph?

Yes, by using composition: https://mui.com/x/react-charts/composition/

The most common use-case is the line chart with bar char. It could also be line and scatter to plot some regression

Reply:
And how would we know which series the `useSeries` is using? 🤔

Reply:
> And how would we know which series the useSeries is using? 🤔

Not sure to understand your question. `useSeries` returns all series. If you want to plot a pie chart in the middle of a bar chart, you technically could with 

```jsx
<ChartsConstainer series={[{tyes: 'pie', ...}, {type: 'bar', ...}]}>
  <BarPlot />
  <PiePot />
</ChartsCOntainer>
```